scripts/dev/compare_output_reports.py

```diff
@@ -0,0 +1,302 @@
+        report_name = report["ReportName"]
+        for table in report["Tables"]:
+            table_name = table["TableName"]
+
+            # Filter out the empty tables, as they do not show up in the SQL
+            if table["Rows"] is None:
+                print(
+                    "Filtering empty JSON table "
+                    f"'{report_name} - {table_name}'"
+                )
+                empty_report_to_tables_list.append((report_name, table_name))
+            else:
+                all_json_report_names.append(report_name)
+                json_report_to_tables_list.append((report_name, table_name))
+
+    return (
+        set(all_json_report_names),
+        set(json_report_to_tables_list),
+        set(empty_report_to_tables_list),
+    )
+
+
+def parse_sql(out_dir: Path):
+    query = """
+    SELECT DISTINCT(ReportName) FROM TabularDataWithStrings
+    """
+
+    abs_sql_path = (out_dir / "eplusout.sql").resolve()
+    sql_uri = "{}?mode=ro".format(abs_sql_path.as_uri())
+
+    with sqlite3.connect(sql_uri, uri=True) as con:
+        cursor = con.cursor()
+        r = cursor.execute(query)
+        all_sql_report_names = [x[0] for x in r.fetchall()]
+
+    query = """
+    SELECT DISTINCT(TableName) FROM TabularDataWithStrings
+    WHERE ReportName = "{}"
+    """
+
+    abs_sql_path = (out_dir / "eplusout.sql").resolve()
+    sql_uri = "{}?mode=ro".format(abs_sql_path.as_uri())
+
+    sql_report_to_tables_list = []
+    for report_name in all_sql_report_names:
+        with sqlite3.connect(sql_uri, uri=True) as con:
+            cursor = con.cursor()
+            r = cursor.execute(query.format(report_name))
+            all_sql_table_names = [x[0] for x in r.fetchall()]
+            for table_name in all_sql_table_names:
+                sql_report_to_tables_list.append((report_name, table_name))
+
+    return set(all_sql_report_names), set(sql_report_to_tables_list)
+
+
+def find_manual_mapping_html():
+    re_predef_report = re.compile(
+        r'= newPreDefReport\(state, "(?P<inReportName>.+?)", "(?P<inReportAbrev>.+?)", "(?P<inReportNamewithSpaces>.+?)"'
+    )
+
+    with open(REPO_ROOT / "src/EnergyPlus/OutputReportPredefined.cc", "r") as f:
+        content = f.read()
+    lines = content.splitlines()
+
+    manual_html_mapping_dict = {}
+    for line in lines:
+        if "= newPreDefReport" in line:
+            m = re_predef_report.search(line)
+            if m:
+                d = m.groupdict()
+                inReportName = d["inReportName"]
+                anchor = MakeAnchorName(d["inReportNamewithSpaces"])
+                if anchor != inReportName:
+                    manual_html_mapping_dict[anchor] = inReportName
+    return manual_html_mapping_dict
+
+
+def parse_html(out_dir):
+    # Lazy load bs4 as its not in the standard library
+    try:
+        from bs4 import BeautifulSoup
+    except ModuleNotFoundError:
+        print("bs4 (BeautifulSoup) is not installed, not checking HTML")
```

Comment:
I haven't checked license and such, but last I checked the bs4 package, it was a single Python file, so we could _potentially_ include it with our repo if we needed it to do our local dev test support.  We already use it with Decent CI where it must be installed on the system.  Definitely a discussion for another day, and I guess the real solution is for devs to just pip install bs4 into whatever build you are testing.  I wonder if we should consider starting a `requirements.txt` that defines some  conditional packages.  For local testing, we could suggest bs4.  For running with ENABLE_REGRESSION_TESTING, you would also want `soupsieve`.  For Decent CI you also want boto, but that's nothing for local devs to worry about.  OK, I'm just rambling at this point, thanks for adding this conditionally.